### ISSUE TITLE
Error on emacs startup

### ISSUE THREAD

**Issue by wedens** [emacs-helm/helm-ls-git#29]
I tried to remove helm/async/helm-ls-git and install them again, it doesn't solve the error (I use MELPA).
Configuration related to helm:
```
(require 'helm-config)
(require 'helm-ls-git)
(helm-mode 1)
(global-set-key (kbd "C-x C-d") 'helm-browse-project)
```
Error:
```
Debugger entered--Lisp error: (void-function eieio-defclass-autoload)
  (eieio-defclass-autoload (quote helm-ls-git-source) (quote (helm-source-in-buffer)) "helm-ls-git" nil)
  eval-buffer(#<buffer  *load*-602879> nil "/home/wedens/.emacs.d/elpa/helm-ls-git-20150402.439/helm-ls-git-autoloads.el" nil t)  ; Reading at buffer position 365
  load-with-code-conversion("/home/wedens/.emacs.d/elpa/helm-ls-git-20150402.439/helm-ls-git-autoloads.el" "/home/wedens/.emacs.d/elpa/helm-ls-git-20150402.439/helm-ls-git-autoloads.el" nil t)
  load("/home/wedens/.emacs.d/elpa/helm-ls-git-20150402.439/helm-ls-git-autoloads" nil t)
  package-activate-1(helm-ls-git [(20150402 439) ((helm (1 5))) "list git files."])
  package-activate(helm-ls-git (20150402 439))
  package-initialize()
  eval-buffer(#<buffer  *load*> nil "/home/wedens/.emacs" nil t)  ; Reading at buffer position 126
  load-with-code-conversion("/home/wedens/.emacs" "/home/wedens/.emacs" t t)
  load("~/.emacs" t t)
```

**Comment by michael-heerdegen** [comment]
AFAIK this is fixed by desinstalling and reinstalling helm, and probably helm-ls-git as well.

**Comment by wedens** [comment]
I tried uninstall+install multiple times. same error.

**Comment by michael-heerdegen** [comment]
Did you restart Emacs in between deinstall and install?  That can be necessary.

**Comment by wedens** [comment]
yes

**Comment by michael-heerdegen** [comment]
I'm not sure why autoloading the defclasses doesn't work in this case, I don't know eieio much.

Adding

```
(require 'eieio)
```

to your .emacs before calling package-initialize should do as a temporary fix, but that should normally not be necessary of course.

**Comment by michael-heerdegen** [comment]
@thierryvolpiatto will surely have a look when he has access to the internet again in some days.

**Comment by wedens** [comment]
It fails in minimal configuration where only helm and helm-ls-git exists.

**Comment by lewang** [comment]
What version of Emacs are you using?

On Mon, Apr 6, 2015 at 10:24 AM, wedens <notifications@github.com> wrote:

> It fails in minimal configuration where only helm and helm-ls-git exists.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/emacs-helm/helm-ls-git/issues/29#issuecomment-90080782>
> .
>



-- 
Le

**Comment by wedens** [comment]
@lewang 24.3.1

**Comment by thierryvolpiatto** [comment]
I have internet access today, not sure I will have an access tomorrow.

This problem with helm-ls-git should be a bad installation/compilation problem
as always, no more.

See #28 it is the same problem.

**Comment by wedens** [comment]
@thierryvolpiatto what can be bad? I removed all other packages, in my .emacs only melpa configuration. I install helm and helm-ls-git packages, restart emacs and this error is shown.

**Comment by michael-heerdegen** [comment]
I think I found it.  It's not a bad installation problem I think.

The problem is that in older Emacs versions, "eieio-defclass-autoload" was not autoloaded.  AFAIK this had been fixed in 24.4.  So, upgrading Emacs should fix this issue.

@wedens Until you upgrade, adding

```
(require 'eieio)
```
to your init file before package-initialize should work.
